splits/readers.py

```diff
@@ -57,8 +57,10 @@ def read(self, num=None):
 
                 if num > 0 and len(val) == num:
                     break
-        except:
+        except StopIteration:
             pass
+        except Exception as e:
+            raise Exception("Unhandled exception in SplitReader:"+repr(e))
```

Comment:
There are at least 2 exceptions that appear to be passed up from the auth handler.

NoAuthHandlerFound("No handler was ready to authenticate. 1 handlers were checked. ['HmacAuthV1Handler'] Check your credentials",)

TypeError('string indices must be integers, not str',)

I'm not confident enough to have a complete list, which is why I'm passing up all errors.